lib/chemistrykit/cli/cli.rb

```diff
@@ -90,63 +52,40 @@ def brew
         # config object injected into the system?
         pass_params if options['params']
 
-        # replace certain config values with run time flags as needed
-        config = override_configs options, config
+        # TODO: expand this to allow for more overrides as needed
+        config.retries_on_failure = options['retry'].to_i if options['retry']
 
         load_page_objects
 
         # get those beakers that should be executed
         beakers = options['beakers'] ? options['beakers'] : Dir.glob(File.join(Dir.getwd, 'beakers/**/*')).select { |file| !File.directory?(file) }
 
-        if options['beakers']
-          # if a beaker(s) defined use them
-          beakers = options['beakers']
-          # if tags are explicity defined, apply them to the selected beakers
-          setup_tags(options['tag'])
-        else
-          # beakers default to everything
-          beakers = Dir.glob(File.join(Dir.getwd, 'beakers/**/*')).select { |file| !File.directory?(file) }
-
-          if options['tag']
-
-            # if tags are explicity defined, apply them to all beakers
-            setup_tags(options['tag'])
-          else
-            # else choose the default tag
-            setup_tags(['depth:shallow'])
-          end
-        end
+        # if tags are explicity defined, apply them to all beakers
+        setup_tags(options['tag'])
 
         # configure rspec
         rspec_config(config)
 
         # based on concurrency parameter run tests
-        if config.concurrency > 1 && ! options['parallel']
-          exit_code = run_in_parallel beakers, config.concurrency, @tags, options
+        if config.concurrency > 1
+          exit_code = run_parallel beakers, config.concurrency
```

Comment:
Instead of creating a new function called run_parallel, how about just calling run_spec with the beakers and config.concurrency. Then move the logic into the run_spec function to determine whether to run in parallel or not.